### PR TITLE
Feature/app secret

### PR DESCRIPTION
**Pull request by PierreJeanjacquot**
:warning: DRAFT : depends on #25 

- prompt the user when running `test` or `deploy` to ask to attach a secret to the app or not
- propose to save choice and secret value (`appSecret` in config file)
- use saved choice and value to avoid too aggressive prompts

When no app secret is set in config => prompt
image
image

When a secret is saved => load from config
image
image

### CODE REVIEW

File: cli/src/cmd/deploy.js [hunk 1]
```diff
@@ -10,30 +10,47 @@ import { readPackageJonConfig } from '../utils/iAppConfigFile.js';
...
-      default: '0.1.0',
-    },
-  ]);
+    const walletAddress = await askForWalletAddress({ spinner });
 
-  const walletAddress = await askForWalletAddress({ spinner });
+    // if an app secret must be set we will need the app owner wallet to push it
+    let iexec;
+    if (appSecret !== null) {
+      const privateKey = await askForWalletPrivateKey({ spinner });
```
Comment (cedric25) [inline cli/src/cmd/deploy.js:40]: Asking for the private key can be a blocker. At least we need to explain why we need it?
Add a spinner info: "To push an app secret, you need to sign a transaction, hence asking for your private key to do it on your behalf."
(Not sure about this exact sentence)

File: cli/src/cli-helpers/askForAppSecret.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+/**
+ * @returns {Promise<string | null>}
+ */
+export async function askForAppSecret({ spinner }) {
+  const config = await readIAppConfig();
+  const { appSecret: savedAppSecret } = config;
+
+  if (savedAppSecret === null) {
+    spinner.log(`App secret is disabled (from "${CONFIG_FILE}")`);
```
[the remark this answers is not in the document]
Comment (PierreJeanjacquot) [reply cli/src/cli-helpers/askForAppSecret.js:12]: using "disabled" was intentional in the sense that the developer answered he wouldn't use an app secret (different from would use but didn't provide one...)

File: cli/README.md [hunk 3]
```diff
@@ -99,6 +101,8 @@ Options:
   the `--protectedData` option followed by the address of the protected data.
 - use `--inputFile <url...>` to provide one or more input files to your iApp
   during run.
+- use `--requesterSecret <key=value...>` to provide one or more requester
+  secrets to your iApp during run.
 
 ## What's next?
```
[the remark this answers is not in the document]
Comment (PierreJeanjacquot) [reply cli/README.md]: added hint in doc 527a702b5bbe789270850fae2bdd9fc2f79537d4 not sure it is sufficient

File: cli/src/cmd/deploy.js (hunk 1 above)
Comment (PierreJeanjacquot) [reply cli/src/cmd/deploy.js:40]: I'm not sure I want to address the problem in this PR, but an option could be to generate a wallet for the users when running `app init` and never ask them to import their wallet. An option for importing wallets for advanced users can be added as well.

File: cli/src/cli-helpers/askForAppSecret.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (PierreJeanjacquot) [reply cli/src/cli-helpers/askForAppSecret.js:12]:
```suggestion
    spinner.log(`"No app secret" is configured (from "${CONFIG_FILE}")`);
```

File: cli/src/cmd/deploy.js (hunk 1 above)
Comment (PierreJeanjacquot) [reply cli/src/cmd/deploy.js:40]: fa38dc0b4a1c694dbd7d6c77596b0d9b179a6058

**Review by Le-Caignec — APPROVED** [review]
- If I understand correctly, the decision has been made to use only one app secret for the developer, right?
- I'm a bit concerned about the fact that everything is stored in plain text in the `iapp.config.json` file. While the prompt hides the secret when the user sets it, this could give the impression that the secret is encrypted, which it is not. Perhaps including a `.gitignore` file in the template to exclude `iapp.config.json` would be a good idea in a new PR may be)

**Comment by PierreJeanjacquot** [comment]
> * If I understand correctly, the decision has been made to use only one app secret for the developer, right?

Yes for now we assume the developer wants to `test` locally and `deploy` the same secret. We may change that following users' feedback.

>     * I'm a bit concerned about the fact that everything is stored in plain text in the `iapp.config.json` file. While the prompt hides the secret when the user sets it, this could give the impression that the secret is encrypted, which it is not. Perhaps including a `.gitignore` file in the template to exclude `iapp.config.json` would be a good idea in a new PR may be)

Same for me, I added `iapp.config.json` in ignore files in PR #34

**Pull request merged**